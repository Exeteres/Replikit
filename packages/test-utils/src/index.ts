export * from "./databaseTestManager";
export * from "./tmp";
export * from "./code";
export * from "./expect";
export * from "./testController";
export * from "./commandTestManager";
export * from "./factories";
