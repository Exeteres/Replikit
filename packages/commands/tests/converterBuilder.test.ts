import { createTestManager } from "@replikit/test-utils";

describe("ConverterBuilder", () => {
    it("should build a converter", () => {
        const { converter } = createTestManager();
        const result = converter(Number)
            .validator(() => void 0)
            .resolver(() => 123)
            .build();
        expect(result).toMatchSnapshot();
    });

    it("should register a constructed converter in a ConverterStorage", () => {
        const { converter, testManager } = createTestManager({
            excludeBasicConverters: true,
            excludeStorageConverters: true
        });
        converter(Number).register();
        expect(testManager.converters).toMatchSnapshot();
    });
});
