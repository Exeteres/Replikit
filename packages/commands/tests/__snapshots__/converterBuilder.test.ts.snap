// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConverterBuilder should build a converter 1`] = `
Object {
  "resolver": [Function],
  "validator": [Function],
}
`;

exports[`ConverterBuilder should register a constructed converter in a ConverterStorage 1`] = `
ConverterStorage {
  "converterMap": Map {
    [Function] => Object {},
  },
}
`;
